### ISSUE TITLE
Migrating from remote event receivers to webhooks

### ISSUE THREAD

**Issue by waaromikniet** [SharePoint/sp-dev-docs#9249]
We are migrating some applications from SharePoint 2019 to SharePoint online. The applications use remote event receivers using synchronous events. I was using this page to see if we can use webhooks. It looks like we can’t migrate to webhooks because they don’t support synchronous events. 

What are alternatives to suport synchronous events?

Will remote event receivers be depracted in the near future? Just like the add-in model and ACS. Or can we still use them in combination with a modern Azure AD App Registration.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: a94cf9ed-0ae1-f5f9-ac2f-eb0ab8244f2c
* Version Independent ID: a94cf9ed-0ae1-f5f9-ac2f-eb0ab8244f2c
* Content: [Transform SharePoint Add-in model Remote Event Receivers to SharePoint Online Webhooks](https://learn.microsoft.com/en-us/sharepoint/dev/sp-add-ins-modernize/from-remote-event-receivers-to-webhooks)
* Content Source: [docs/sp-add-ins-modernize/from-remote-event-receivers-to-webhooks.md](https://github.com/SharePoint/sp-dev-docs/blob/main/docs/sp-add-ins-modernize/from-remote-event-receivers-to-webhooks.md)
* Service: **sharepoint**
* GitHub Login: @VesaJuvonen
* Microsoft Alias: **vesaj**

**Comment by andrewconnell** [comment]
I'll let Microsoft comment on the future of RER's.

From my POV... RER's are old tech (like the add-in model & ACS) & the lack of investments in that area, with investments in other areas, doesn't necessarily imply a positive future for them.

To directly answer your question, there's no sync way to implement RER's with webhooks. Webhooks simply tell you that something happened, they don't expect a response.

**Comment by waaromikniet** [comment]
@andrewconnell  thanks for the reply. Is there a change webhooks will support sync events in the future? 
Based on the path Microsoft is going with this 'old tech' It looks like we have to rethink our migration path.

Maybe @VesaJuvonen can say something about the future of remote event receivers.

**Comment by andrewconnell** [comment]
@waaromikniet 
> Is there a chance webhooks will support sync events in the future?

No... generally speaking (_across the industry, not just a MSFT/SharePoint position_) webhooks aren't sync nor do they process responses more generic than success/failed. They are simply used for an app to notify another endpoint that something happened. Sometimes they include data, sometimes they just mention the basic "something happened that you wanted to be notified about.

While Microsoft hasn't said anything about what they'd add to webhooks in the future, I don't expect it.
